src/shared/components/VideoPlayer/VideoPlayer.tsx

```diff
@@ -115,37 +158,201 @@ const VideoPlayerComponent: React.ForwardRefRenderFunction<HTMLVideoElement, Vid
     player.play()
   }
 
+  useEffect(() => {
+    if (!player) {
+      return
+    }
+    const handler = () => setVideoTime(Math.floor(player.currentTime()))
+
+    player.on('timeupdate', handler)
+    return () => {
+      player.off('timeupdate', handler)
+    }
+  }, [player])
+
+  useEffect(() => {
+    if (!player) {
+      return
+    }
+    const handler = () => {
+      if (player.isFullscreen()) {
+        setIsFullScreen(true)
+      } else {
+        setIsFullScreen(false)
+      }
+    }
+    player.on('fullscreenchange', handler)
+    return () => {
+      player.off('fullscreenchange', handler)
+    }
+  }, [player])
+
+  useEffect(() => {
+    if (!player) {
+      return
+    }
+    const handler = () => {
+      if (player.isFullscreen()) {
+        setIsFullScreen(true)
+      } else {
+        setIsFullScreen(false)
+      }
+    }
+    player.on('fullscreenchange', handler)
+    return () => {
+      player.off('fullscreenchange', handler)
+    }
+  }, [player])
+
+  useEffect(() => {
+    if (!player) {
+      return
+    }
+    const enterPiPhandler = () => {
+      setIsPiPEnabled(true)
+    }
+    const exitPiPhandler = () => {
+      setIsPiPEnabled(false)
+    }
+
+    player.on('enterpictureinpicture', enterPiPhandler)
+    player.on('leavepictureinpicture', exitPiPhandler)
+    return () => {
+      player.off('enterpictureinpicture', enterPiPhandler)
+      player.off('leavepictureinpicture', exitPiPhandler)
+    }
+  }, [player])
+
   const debouncedVolumeChange = useRef(
     debounce((volume: number) => {
-      updatePlayerVolume(volume)
+      updateCachedPlayerVolume(volume)
     }, 500)
   )
 
-  const isInitialMount = useRef(true)
+  // update volume on keyboard input
   useEffect(() => {
-    if (!player || !isInitialMount) {
+    if (!player) {
       return
     }
-    isInitialMount.current = false
-
-    player.volume(playerVolume)
 
-    const handleVolumeChange = () => debouncedVolumeChange.current(player.volume())
-    player.on('volumechange', handleVolumeChange)
+    const handler = (e: Event) => {
```

Comment:
Use meaningful names
```suggestion
    const handler = (event: Event) => {
```